### PR TITLE
asm app gateways commands implemented

### PR DESCRIPTION
**Pull request by MikhailTryakhov**
ASM app gateways commands to manage frontend IPs, frontend ports, http listeners and load balancing rules are implemented

### CODE REVIEW

**Comment by anuchandy** [comment]
@MikhailTryakhov please see above comments, since we have  vpn gateway unblocked, you might want to test it, then we can do final review, please synch your branch MSOpenTech:dev-asm-network.

**Comment by MikhailTryakhov** [comment]
@anuchandy Please review this PR

**Comment by MikhailTryakhov** [comment]
Attaching full http request/response: https://gist.github.com/MikhailTryakhov/0d00525dc1f075d54803

File: lib/util/utils.js [hunk 1]
```diff
@@ -1185,7 +1185,7 @@ exports.findFirstCaseIgnore = function (list, attributes) {
  * case-insensitive.
  */
 exports.indexOfCaseIgnore = function (list, attributes) {
-  if (_.isEmpty(attributes)) {
+  if (!list || list.length === 0 || _.isEmpty(attributes)) {
```
Comment (romangromov) [inline lib/util/utils.js]: Looks like this case is already handled inside `_.isEmpty` check

**Comment by MikhailTryakhov** [comment]
@anuchandy Comments are fixed. Could you please take a look again?

**Comment by anuchandy** [comment]
LGTM thanks @MikhailTryakhov

**Pull request merged**